Tests/backend-specific-tests/Redshift/DataLoaderPersistentRedshiftWorkspaceTest.php

```diff
@@ -268,30 +350,40 @@ public function testRedshiftWorkspaceConfigMultipleWorkspace()
 
         $clientWrapper = new ClientWrapper($client, null, null, '');
         $logger = new TestLogger();
-        try {
-            $workspaceFactory = new WorkspaceProviderFactoryFactory(
-                $logger,
-                $clientWrapper
-            );
-            $workspaceFactory->getWorkspaceProviderFactory(
-                'workspace-redshift',
-                $component,
-                $configurationId
-            );
-        } catch (ApplicationException $e) {
-            self::assertEquals(
-                sprintf(
-                    'Multiple workspaces (total 2) found (IDs: %s, %s) for configuration "%s" of component "%s".',
-                    $workspace1['id'],
-                    $workspace2['id'],
-                    $configurationId,
-                    'keboola.runner-config-test'
-                ),
-                $e->getMessage()
-            );
-            $workspacesApi = new Workspaces($this->client);
-            $workspacesApi->deleteWorkspace($workspace1['id']);
-            $workspacesApi->deleteWorkspace($workspace2['id']);
-        }
+
+        $dataLoader = new DataLoader(
+            $clientWrapper,
+            $logger,
+            $this->workingDir->getDataDir(),
+            new JobDefinition([], $component, $configurationId),
+            new OutputFilter()
+        );
+        $dataLoader->storeOutput();
+
+        $this->assertTrue($logger->hasWarning(
+            sprintf(
+                'Multiple workspaces (total 2) found (IDs: %s) for configuration "%s" of component "%s", using "%s".',
+                implode(',', [$workspace1['id'], $workspace2['id']]),
+                $configurationId,
+                'keboola.runner-config-test',
+                $workspace1['id']
```

Comment:
see `rsort` comment bellow.  I would have guessed this to be `$workspace2['id']`